source/swap/contracts/Swap.sol

```diff
@@ -74,6 +74,10 @@ contract Swap is ISwap {
     Types.Order calldata order
   ) external {
 
+    // Ensure signatureVerifier defined in order is this contract
+    require(order.signatureVerifier == address(this),
+      "INVALID_SWAP_ADDRESS");
```

Comment:
I guess if we're naming the variable `signatureVerifier`, then this revert message should change too

Reply:
removing the check per dons's comment: "it doesn't need to be checked because the contract address is in the domainSeparator"